### PR TITLE
storage: split window PoST submission into multiple messages

### PR DESCRIPTION
**Pull request by dirkmc**
Fixes https://github.com/filecoin-project/lotus/issues/3332

### CODE REVIEW

File: storage/wdpost_run.go [hunk 1]
```diff
@@ -324,116 +319,167 @@ func (s *WindowPoStScheduler) runPost(ctx context.Context, di miner.DeadlineInfo
...
-		Partitions: make([]miner.PoStPartition, 0, len(partitions)),
-		Proofs:     nil,
+	// Split partitions into batches, so as not to exceed the number of sectors
+	// allowed in a single message
+	partitionBatches, err := s.batchPartitions(partitions)
+	if err != nil {
+		return nil, err
 	}
 
-	var sinfos []proof.SectorInfo
```
Comment (dirkmc) [inline storage/wdpost_run.go:338]: The logic here should be the same, the indentation just makes it hard to compare

**Comment by dirkmc** [comment]
Note: TestAPIRPC/testMiningReal failed in CI but works for me locally

File: storage/wdpost_run.go [hunk 2]
```diff
@@ -324,116 +319,167 @@ func (s *WindowPoStScheduler) runPost(ctx context.Context, di miner.DeadlineInfo
...
+
+	// The number of messages will be:
+	// ceiling(number of partitions / partitions per message)
+	batchCount := uint64(math.Ceil(float64(len(partitions)) / float64(partitionsPerMsg)))
+
+	// Split the partitions into batches
+	batches := make([][]*miner.Partition, 0, batchCount)
+	for i := 0; i < len(partitions); {
+		batch := make([]*miner.Partition, 0, partitionsPerMsg)
+		for j := 0; j < partitionsPerMsg && i < len(partitions); j++ {
```
Comment (Stebalien) [inline storage/wdpost_run.go:476]: Why not just slice?

**Review by Stebalien — APPROVED** [review]
Everything LGTM except for the moved randomness.

**Review by Stebalien — APPROVED** [review]

**Comment by dirkmc** [comment]
@Stebalien sorry I had to rebase again after https://github.com/filecoin-project/lotus/pull/3710 was merged, please take one more look

File: storage/wdpost_run.go [hunk 3]
```diff
@@ -326,136 +320,200 @@ func (s *WindowPoStScheduler) runPost(ctx context.Context, di dline.Info, ts *ty
...
 
-	skipCount := uint64(0)
-	postSkipped := bitfield.New()
-	var postOut []proof.PoStProof
+	// Generate proofs in batches
+	posts := make([]miner.SubmitWindowedPoStParams, 0, len(partitionBatches))
+	for batchIdx, batch := range partitionBatches {
+		batchPartitionStartIdx := 0
+		for _, batch := range partitionBatches[:batchIdx] {
+			batchPartitionStartIdx += len(batch)
```
Comment (Stebalien) [inline storage/wdpost_run.go:341]: Why not just keep a running count?

**Review by Stebalien — APPROVED** [review]
Nice!

**Review by magik6k — APPROVED** [review]

**Pull request merged**